checkov/arm/checks/resource/SQLServerNoPublicAccess.py

```diff
@@ -24,7 +24,9 @@ def scan_resource_conf(self, conf):
                                 resource["type"] == "firewallrules":
                             if "properties" in resource:
                                 if "startIpAddress" in resource["properties"] and \
-                                        resource["properties"]["startIpAddress"] in ['0.0.0.0', '0.0.0.0/0']:  # nosec
+                                        resource["properties"]["startIpAddress"] in ['0.0.0.0', '0.0.0.0/0'] and \
+                                            "endIpAddress" in resource["properties"] and \
+                                                resource["properties"]["endIpAddress"] == '255.255.255.255':  # nosec
```

Comment:
also here the `# nosec` needs to be on the first line and brackets around the whole statement would be great :)
```python
    if "properties" in resource:
        if ("startIpAddress" in resource["properties"] and  # nosec
                resource["properties"]["startIpAddress"] in ['0.0.0.0', '0.0.0.0/0'] and
                "endIpAddress" in resource["properties"] and
                resource["properties"]["endIpAddress"] == '255.255.255.255'):
            return CheckResult.FAILED
```

Reply:
Thanks, I've used your suggested code directly in b38e93e!